tests/v2/validation/provisioning/k3s/hardened_cluster_test.go

```diff
@@ -0,0 +1,135 @@
+	require.NoError(c.T(), err)
+
+	enabled := true
+	var testuser = namegen.AppendRandomString("testuser-")
+	var testpassword = password.GenerateUserPassword("testpass-")
+	user := &management.User{
+		Username: testuser,
+		Password: testpassword,
+		Name:     testuser,
+		Enabled:  &enabled,
+	}
+
+	newUser, err := users.CreateUserWithRole(client, user, "user")
+	require.NoError(c.T(), err)
+
+	newUser.Password = user.Password
+
+	standardUserClient, err := client.AsUser(newUser)
+	require.NoError(c.T(), err)
+
+	c.standardUserClient = standardUserClient
+}
+
+func (c *HardenedK3SClusterProvisioningTestSuite) TestProvisioningK3SHardenedCluster() {
+	nodeRolesDedicated := []provisioninginput.MachinePools{provisioninginput.EtcdMachinePool, provisioninginput.ControlPlaneMachinePool, provisioninginput.WorkerMachinePool}
+
+	tests := []struct {
+		name            string
+		client          *rancher.Client
+		machinePools    []provisioninginput.MachinePools
+		scanProfileName string
+	}{
+		{"CIS 1.8 Profile Hardened " + provisioninginput.StandardClientName.String(), c.standardUserClient, nodeRolesDedicated, "k3s-cis-1.8-profile-hardened"},
+		{"CIS 1.8 Profile Permissive " + provisioninginput.StandardClientName.String(), c.standardUserClient, nodeRolesDedicated, "k3s-cis-1.8-profile-permissive"},
+	}
+	for _, tt := range tests {
+		c.Run(tt.name, func() {
+			provisioningConfig := *c.provisioningConfig
+			provisioningConfig.MachinePools = tt.machinePools
+			provisioningConfig.Hardened = true
+
+			nodeProviders := provisioningConfig.NodeProviders[0]
+			externalNodeProvider := provisioning.ExternalNodeProviderSetup(nodeProviders)
+
+			testConfig := clusters.ConvertConfigToClusterConfig(&provisioningConfig)
+			testConfig.KubernetesVersion = c.provisioningConfig.K3SKubernetesVersions[0]
+
+			clusterObject, err := provisioning.CreateProvisioningCustomCluster(tt.client, &externalNodeProvider, testConfig)
+			require.NoError(c.T(), err)
+
+			provisioning.VerifyCluster(c.T(), tt.client, testConfig, clusterObject)
+
+			cluster, err := clusters.NewClusterMeta(tt.client, clusterObject.Name)
+			require.NoError(c.T(), err)
+
+			latestCISBenchmarkVersion, err := tt.client.Catalog.GetLatestChartVersion(charts.CISBenchmarkName, catalog.RancherChartRepo)
+			require.NoError(c.T(), err)
+
+			project, err := projects.GetProjectByName(tt.client, cluster.ID, cis.System)
+			require.NoError(c.T(), err)
+
+			c.project = project
+			require.NotEmpty(c.T(), c.project)
+
+			c.chartInstallOptions = &charts.InstallOptions{
+				Cluster:   cluster,
+				Version:   latestCISBenchmarkVersion,
+				ProjectID: c.project.ID,
+			}
+
+			cis.SetupCISBenchmarkChart(tt.client, c.project.ClusterID, c.chartInstallOptions, charts.CISBenchmarkNamespace)
+			cis.RunCISScan(tt.client, c.project.ClusterID, tt.scanProfileName)
```

Comment:
I don't think this matters but, it might be worth running another verifyCluster after the scan is complete to ensure the chart and other pods are still healthy.

Reply:
Yeah I don't think that's necessary. It doesn't impact that at all.